lib/mpl_toolkits/mplot3d/proj3d.py

```diff
@@ -128,15 +151,20 @@ def ortho_transformation(zfront, zback):
                      [0, 0, a, b]])
 
 
-def proj_transform_vec(vec, M):
+def _proj_transform_vec(vec, M):
     vecw = np.dot(M, vec)
     w = vecw[3]
     # clip here..
     txs, tys, tzs = vecw[0]/w, vecw[1]/w, vecw[2]/w
     return txs, tys, tzs
 
 
-def proj_transform_vec_clip(vec, M):
+@cbook.deprecated("3.1")
+def proj_transform_vec(vec, M):
+    return _proj_transform_vec(vec, M)
+
+
+def _proj_transform_vec_clip(vec, M):
```

Comment:
These deprecations might be a bit problematic. IIRC, some of these functions in proj3d.py are used elsewhere in 3rd party tools like mplcursors.

Reply:
particularly, the thing I am thinking of is any code that tries to infer the 3d coordinates from a 2D location, so that coordinates can be displayed in the interactive text display in the lower right-hand corner of the figure window.

Reply:
yeah, `line2d_seg_dist()` is what is used.